include/glow/Backends/CompileOptions.h

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_BACKENDS_COMPILEOPTIONS_H
+#define GLOW_BACKENDS_COMPILEOPTIONS_H
+
+namespace glow {
+
+enum class CompilationMode {
+  Train, /// Compile the graph in preperation for training.
+  Infer, /// Compile the graph for inference. Notice that this operation
+         /// changes the graph in a way that is not reversible.
+};
+
+/// Configuration options for the compile() method on Backend.
+struct CompileOptions {
```

Comment:
We have `CompilationMode`, but `CompileOptions`. This looks odd. Should we use a more consistent naming?

Reply:
Nit: Why do we have this header in the Backends subdirectory? Is it the most logical place to put it?

Reply:
What's your preferred alternative?

Reply:
For naming, I'd go with `CompilationMode` and `CompilationOptions`.

For the subdirectory, I don't have any strong opinion. May be a new `include/glow/Options/` subdirectory?

Reply:
CompilationOptions sounds fine to me but creating a new subdirectory for a single header seems overkill, what else would we need to put in there?

Reply:
> but creating a new subdirectory for a single header seems overkill, what else would we need to put in there?

Yeah, I agree. My initial (very small) concern was that the options are not for backends only. They are also used by optimizers, etc.